tools/schemapi/schemapi.py

```diff
@@ -555,6 +565,7 @@ def _passthrough(*args, **kwds):
                         dct,
                         schema=possible_schema,
                         rootschema=rootschema,
+                        default_class=default_class,
```

Comment:
To localize the logic a bit, I'd remove lines 550-552 and write `default_class=cls or default_class` here.

Reply:
I implemented about half of these changes, make `default_class` default to `_passthrough`.  (To get that to work I had to move the definition of `_passthrough` to the outside.)  From my understanding of the logic, we aren't able to use `default_class=cls or default_class`, because `cls` might get changed earlier, and I think we only want to use `cls` here if it arrived as an input argument.